### ISSUE TITLE
Spec formatting interacts poorly with browser text manipulation features / find on page

### ISSUE THREAD

**Issue by ianh** [WebAssembly/spec#1515]
For example, try searching for the text 0x00 on the page <https://webassembly.github.io/spec/core/binary/instructions.html>.  In Safari, that string is "Not found" -- in Chrome, there are seven results, but the correct result is never indicated using the normal selection highlight, and the browser doesn't even scroll to the right point for all the results.  Attempting to select the text 0x00 doesn't work at all in Chrome.  In Safari, attempting to select text does something difficult for me to describe other than that it's "slightly cursed".

Anyone working directly with WebAssembly will be paging through these documents constantly, so it would be helpful for them to support basic browser text operations.  Or, at least, I've personally been annoyed by these issues enough to file this issue, and I'm sure I'm not the only one!

**Comment by rossberg** [comment]
Yes, I understand, and I share the sentiment. Unfortunately, this is a result of the poor state of rendering math on the Web and the glorious hack that is MathJax to work around it. And unfortunately, MathJax 3 dropped support for switching to MathML output, even on browsers that support it.

You could try the single page W3C version (e.g., https://webassembly.github.io/spec/core/bikeshed/#instructions⑦), which is rendered with Katex instead. That has its own problems, but at least allows more functional text search and C&P.

Alternatively, the PDF version should not have these issues, and looks nicer, too. :)

**Comment by 7ombie** [comment]
Why can't the docs use ASCII syntax? We're programmers, not physicists.

(Apple) Safari can't even align latin-lowers correctly:

<img width="839" alt="Screen Shot 2022-08-17 at 21 34 15" src="https://user-images.githubusercontent.com/80077603/185237880-80c02162-fbb6-4bee-bf82-57a68929c2d1.png">

**Comment by tomayac** [comment]
@rossberg Luckily (Chrome as the last browser 🙈) now [supports MathML](https://caniuse.com/mathml) directly, so there might be a chance to switch to MathML directly?! This was just raised by @MendyBerger in the context of the [WebAssembly.org makeover](https://github.com/WebAssembly/meetings/blob/main/main/2024/CG-06.md#wednesday-june-5:~:text=WebAssembly.org%20makeover%20(Tom%20Steiner)) session at the June 2024 CG meeting.

**Comment by rossberg** [comment]
There is additional context on issue #1141. In particular, last time I tried this about a year ago:

* Rendering on Chrome worked but the result was [still very broken](https://github.com/WebAssembly/spec/issues/1141#issuecomment-1500910665); that might have improved over the last year. I don't know how well other browsers do.

* The size of the HTML blew up by 30x (!); that is [almost entirely](https://github.com/WebAssembly/spec/issues/1141#issuecomment-1501774837) due to the inherent space inefficiency of HTML syntax and its lack of macro capabilities.

The latter probably means that we will never want to statically generate MathML, since the whole doc would easily blow up to >0.1 GB that way. However, we could try to [configure MathJax](https://docs.mathjax.org/en/v3.0-latest/output/mathml.html) to render client-side with MathML. If somebody is eager to try that, please go ahead!

**Comment by tomayac** [comment]
There's apparently also a way to [make the content copyable and searchable via a visually hidden element](https://groups.google.com/g/mathjax-users/c/1BzTQFINbqY/m/xGYAlhDNAwAJ?pli=1). This via https://github.com/mathjax/MathJax/issues/2240#issuecomment-857091474, which I looked up to see if there's a solution at the MathJax level.

**Comment by valadaptive** [comment]
This isn't even just about "rendering math"--many phrases and operation names are rendered using mathematical symbols instead of standard characters. For instance, if you want to search for the "bitmask" operation on [the Instructions page](https://webassembly.github.io/spec/core/exec/instructions.html), you can't do Ctrl+F "bitmask". The actual text you'd need to search is "𝖻𝗂𝗍𝗆𝖺𝗌𝗄". That's "Mathematical Sans-Serif Small B", "Mathematical Sans-Serif Small I", etc. This is *incredibly* annoying.
